app/views/maintenance_tasks/runs/_run.html.erb

```diff
@@ -8,4 +8,7 @@
   <% else %>
     <%= render 'maintenance_tasks/runs/info', run: run, with_status: true %>
   <% end %>
+  <% if defined?(task) && task.csv_task? %>
```

Comment:
This is not so nice, but a result of the fact that this partial is shared between the previous runs section on the show page, and the runs index page (in which Runs are not scoped to a Task). It seems like from https://github.com/Shopify/maintenance_tasks/pull/288 we might end up getting rid of this anyways, which would simplify our lives here because we could rely on `task` being defined.

Reply:
Alternatively, we could map runs to TaskData objects in `RunsController#index`, but I think that defeats the purpose of having the separate page and adds extra complexity I'm not sure we want... But, being able to download CSVs for deleted Tasks was on our list of advantages for keeping the full list of runs, s maybe we do want it.

Reply:
Why not check whether the run itself has an attachment instead of looking up the Task? Something like `run.csv_file.attached?`. We may need to preload that relation though.

Reply:
Oh yeah good call, I thought we'd need to download and check if it was nil, but we can definitely use `attached?` instead